packages/public/inspection-capture-web/src/PhotoCaptureHUD/PhotoCaptureHUDAddDamagePreview/PhotoCaptureHUDAddDamagePreview.styles.ts

```diff
@@ -0,0 +1,26 @@
+import { Styles } from '@monkvision/types';
+
+export const styles: Styles = {
+  container: {
+    position: 'absolute',
+    display: 'flex',
+    alignSelf: 'stretch',
+    justifyContent: 'center',
+    flexDirection: 'column',
+    flex: '1',
+  },
+  top: {
+    position: 'absolute',
+    display: 'flex',
+    alignSelf: 'flex-start',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    margin: '10px',
+    zIndex: '1',
+    top: '0',
+    right: '0',
+  },
+  addDamageButton: {
+    backgroundColor: 'rgba(52, 53, 63, 0.64)',
```

Comment:
Use a color from the theme instead (you don't need to have the EXACT same color as the one in the Figma, juste take whatever color is the closest in the default palette). If you want to change the opacity of the color you can use the utility function that does this in the common package

Reply:
I added it directly to the component like this:
```
export function AddDamageButton({ onAddDamage }: AddDamageButtonProps) {
  const { t } = useTranslation();
  const { palette } = useMonkTheme();

  const bgColor = getHexFromRGBA({ ...getRGBAFromString(palette.secondary.xdark), a: 0.64 });

 return (
    <Button
      icon='add'
      onClick={() => onAddDamage(HUDMode.ADD_DAMAGE)}
      data-testid='monk-test-btn'
      style={{ backgroundColor: bgColor }}
    >
      {t('photo.hud.sight.addDamageBtn')}
    </Button>
  );
}
```
Should I create a hook file to handle this for every component ?

Reply:
Okay so three things :
- First, know that the two util functions `getHexFromRGBA` and `getRGBAFromString` are a tiny bit expensive so whenever you use them (or use another function that uses them), I think it's better to memoize them (put them in `useMemo`). So in your case it would mean something like that :
```typescript
const bgColor = useMemo(
  () => getHexFromRGBA({ ...getRGBAFromString(palette.secondary.xdark), a: 0.64 }),
  [palette],
);
```
- Second : in [this PR](https://github.com/monkvision/monkjs/pull/656), I created a util function called `changeAlpha` that changes the alpha value of a color, which is exactly what you did here. Since I already wrote the function, the doc and the tests in the PR, I suggest we merge the PR, and then you simply rebase the changes onto your branch and use the util function instead of manually changing the alpha value.
- Third : For your custom hook suggestion, if you are using the exact same alpha change for every semi-transparent button in the HUD, you could definitely create a custom hook called `usePhotoHUDButtonBackground` that will do two things :
  - Change the alpha of the color from the palette and return the proper background color
  - Make sure to memoize the call to `changeAlpha` since its an expensive function

What do you think?